onadata/apps/api/tests/viewsets/test_xform_viewset.py

```diff
@@ -3350,6 +3352,13 @@ def test_different_form_versions(self):
                 request = self.factory.patch('/', data=post_data, **self.extra)
                 response = view(request, pk=self.xform.pk)
                 self.assertEqual(response.status_code, 200)
+                # ensure that an XForm version object is created
+                versions = XFormVersion.objects.filter(xform=self.xform)
+                self.assertEqual(versions.count(), 1)
```

Comment:
Yes; I'll make changes to create an `XFormVersion` object on form uploads. Ideally we'd only need to check the `XFormVersion` for the versions of the form; this may not be the case for forms uploaded previously before the changes... for those forms we'd need to query the XForm instances.

Reply:
We can handle that via a migration, by creating for all existing forms. What I had concerns about is reprocessing XLSForm that had been uploaded, I do not think we should reprocess XLSForms but we can create versions of existing forms.